legion_core/src/storage.rs

```diff
@@ -746,7 +746,26 @@ impl ArchetypeData {
         self.tags.validate(self.chunk_sets.len());
     }
 
-    pub(crate) fn enumerate_entities<'a>(
+    /// Iterate all entities in existence by iterating across archetypes, chunk sets, and chunks
+    pub(crate) fn iter_entities<'a>(&'a self) -> impl Iterator<Item = Entity> + 'a {
+        self.chunk_sets
+            .iter()
+            .enumerate()
```

Comment:
I don't think the `enumerate` calls are necessary here given the index in every closure is unused.